web/js/map-OpenLayers.js

```diff
@@ -375,6 +375,66 @@ $.extend(fixmystreet.utils, {
               new OpenLayers.LonLat( state.lon, state.lat ),
               state.zoom
           );
+      },
+
+      setup_geolocation: function() {
+          if (!OpenLayers.Control.Geolocate || !fixmystreet.map ||
+              !fixmystreet.utils || !fixmystreet.utils.parse_query_string ||
+              fixmystreet.utils.parse_query_string().geolocate !== '1'
+          ) {
+              return;
+          }
+
+          var layer;
+          function addGeolocationLayer() {
+            layer = new OpenLayers.Layer.Vector('Geolocation');
+            fixmystreet.map.addLayer(layer);
+            layer.setZIndex(fixmystreet.map.getLayersByName("Pins")[0].getZIndex() - 1);
+          }
+
+          function updateGeolocationMarker(e) {
+              if (!layer) {
+                  addGeolocationLayer();
+              }
+              layer.removeAllFeatures();
```

Comment:
Could we move the features instead of recreating them every time there's a location update?

Reply:
Just to flog a dead horse just a little more, if the blue dot is permanent, rather than constantly being recreated, then that would open the door to a putting a CSS "pulse" animation onto it…